### PR TITLE
[SPARK-10084] [MLlib] [Doc] Add Python example for mllib FP-growth user guide

### PR DESCRIPTION
**Pull request by yanboliang**
1, Add Python example for mllib FP-growth user guide.
2, Correct mistakes of Scala and Java examples.

### CODE REVIEW

File: docs/mllib-frequent-pattern-mining.md [hunk 1]
```diff
@@ -41,21 +41,23 @@ MLlib's FP-growth implementation takes the following (hyper-)parameters:
...
-import org.apache.spark.mllib.fpm.{FPGrowth, FPGrowthModel}
+import org.apache.spark.mllib.fpm.FPGrowth
 
-val transactions: RDD[Array[String]] = ...
+val data = sc.textFile("data/mllib/sample_fpgrowth.txt")
 
-val fpg = new FPGrowth()
+val transactions: RDD[Array[String]] = data.map(s => s.trim.split(' '))
+
+val fpm = new FPGrowth()
```
Comment (mengxr) [inline docs/mllib-frequent-pattern-mining.md]: Should be `fpg` for `FP-growth`.

File: docs/mllib-frequent-pattern-mining.md [hunk 2]
```diff
@@ -77,22 +79,60 @@ that stores the frequent itemsets with their frequencies.
 import java.util.List;
 
 import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
```
Comment (mengxr) [inline docs/mllib-frequent-pattern-mining.md]: Guava utils are not necessary here. Could you help verify the following?

1. The `Joiner` usage can be removed. `println("[" + Joiner.on(",").join(itemset.javaItems()) + "]")` is the same as `println(itemset.javaItems())`.
2. `Lists.newArrayList` could be replaced by Java's `Arrays.asList`.

Then we can remove Guava dependency from the example code.

**Comment by yanboliang** [comment]
Jenkins, test this please.

**Comment by yanboliang** [comment]
Jenkins, test this please.

**Comment by mengxr** [comment]
@yanboliang Could you rebase master? It has merge conflicts with #8207.

File: docs/mllib-frequent-pattern-mining.md [hunk 3]
```diff
@@ -133,6 +134,36 @@ for (AssociationRules.Rule<String> rule
...
+
+<div data-lang="python" markdown="1">
+
+[`FPGrowth`](api/python/pyspark.mllib.html#pyspark.mllib.fpm.FPGrowth) implements the
+FP-growth algorithm.
+It take an `RDD` of transactions, where each transaction is an `List` of items of a generic type.
+Calling `FPGrowth.train` with transactions returns an
+[`FPGrowthModel`](api/python/pyspark.mllib.html#pyspark.mllib.fpm.FPGrowthModel)
+that stores the frequent itemsets with their frequencies. The following
+example illustrates how to mine frequent itemsets and association rules
```
Comment (mengxr) [inline docs/mllib-frequent-pattern-mining.md]: `Association rules` is not available in Python yet.

File: docs/mllib-frequent-pattern-mining.md [hunk 4]
```diff
@@ -133,6 +134,36 @@ for (AssociationRules.Rule<String> rule
...
+details) from `transactions`.
+
+{% highlight python %}
+from pyspark.mllib.fpm import FPGrowth
+
+data = sc.textFile("data/mllib/sample_fpgrowth.txt")
+
+transactions = data.map(lambda line: line.strip().split(' '))
+
+model = FPGrowth.train(transactions, 0.2, 10)
```
Comment (mengxr) [inline docs/mllib-frequent-pattern-mining.md]: Use named arguments.

**Comment by mengxr** [comment]
LGTM. Merged into master and branch-1.5. Thanks! (Skipped Jenkins because this is doc-only.)

**Pull request closed without merging**